design-proposals/storage-class-migration/storage-class-migration.md

```diff
@@ -0,0 +1,269 @@
+# Overview
+This design proposal extends KubeVirt APIs to enable the migration between storage classes.
+
+## Motivation
+In some circumstances, the guest owner may want to have the option of switching the storage provider or class for a group of volumes used by an active VM. The VM must currently be turned off for this operation due to service disruptions.
+
+
+## Goals
+Design and develop an API to handle storage migrations.
+
+ ## Non  Goals
+ This feature doesn't cover the VM Live migration with RWO volumes.
+ 
+ ## Definition of Users
+ This feature addresses for all KubeVirt users who want to move to another storage classes without workload interuptions.
+ 
+## User Stories
+  1. A storage class of volumes used by a running VM is deprecated
+  2. A more performant storage class is now available in the cluster and the    user wants start using it.
```

Comment:
another use-case is that a storage device is filling up

Reply:
Added